bindings_ffi/src/lib.rs

```diff
@@ -302,6 +323,44 @@ mod tests {
         assert!(!client.wallet_address().is_empty());
     }
 
+    #[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+    async fn test_create_client_with_storage() {
+        let ffi_inbox_owner = LocalWalletInboxOwner::new();
+
+        let path = TempPath::from_path("./ffidb.db3");
+
+        let client_a = create_client(
+            Box::new(MockLogger {}),
+            Box::new(ffi_inbox_owner.clone()),
+            xmtp_networking::LOCALHOST_ADDRESS.to_string(),
+            false,
+            Some(path.to_string_lossy().to_string()),
+        )
+        .await
+        .unwrap();
+
+        let installation_id = client_a.inner_client.installation_id();
+        drop(client_a);
+
+        let client_b = create_client(
+            Box::new(MockLogger {}),
+            Box::new(ffi_inbox_owner),
+            xmtp_networking::LOCALHOST_ADDRESS.to_string(),
+            false,
+            Some(path.to_string_lossy().to_string()),
+        )
+        .await
+        .unwrap();
+
+        let other_installation_id = client_b.inner_client.installation_id();
+        drop(client_b);
+
+        assert!(
+            installation_id == other_installation_id,
+            "did not use same installation ID"
+        )
```

Comment:
I wasn't sure if there was a better thing to assert here, but this test does fail if you don't pass a location for the local DB so I think it's at least verifying something.

Reply:
If you'd like, you can send a message from Client A, and then list messages on Client B, and verify that the sent message is in the list (the DB is used to store it)